lib/ansible/modules/remote_management/manageiq/manageiq_dynamic_resource_definition.py

```diff
@@ -0,0 +1,283 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) Ahmed Bashir <ahmwag@gmail.com> @ahmbas
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+
+module: manageiq_dynamic_resource_definition
+
+short_description: Management of dynamic resource definitions in ManageIQ.
+extends_documentation_fragment: manageiq
+version_added: '2.6'
+author: Ahmed Bashir (@ahmbas)
+description:
+  - The manageiq_dynamic_resource_definition module supports adding, updating and deleting dynamic resource definitions in ManageIQ.
+
+options:
+  state:
+    description:
+      - absent - dynamic resource definitions should not exist, present - dynamic resource definitions should be.
+    choices: ['absent', 'present']
+    default: 'present'
+  name:
+    description:
+      - The dynamic resource definition's name.
+  properties:
+    description:
+      - The dynamic resource definition's properties.
```

Comment:
please document what this consists of.  you may use "suboptions:" (see manageiq_provider for example), or just prose.

hmm, do you need this nesting at all?
how about this module taking `attributes:` and `associations:` args at top level?